src/hotspot/share/runtime/flags/jvmFlagAccess.cpp

```diff
@@ -66,6 +66,9 @@ class TypedFlagAccessImpl : public FlagAccessImpl {
     if (constraint != nullptr && constraint->phase() <= static_cast<int>(JVMFlagLimit::validating_phase())) {
       JVMFlag::Error err = typed_check_constraint(constraint->constraint_func(), value, verbose);
       if (err != JVMFlag::SUCCESS) {
+        if (origin == JVMFlagOrigin::ERGONOMIC) {
+          fatal("FLAG_SET_ERGO cannot be used to set an invalid value for %s", flag->name());
```

Comment:
This would be okay if we are guaranteed to see the `printError` message from the constraint function, but I don't think we are. If I'm reading things correctly we need `verbose` to be true to get the error message, but `verbose` comes from `AtParse` which seems to be set to 0 and never modified. ??
I think we need to force `verbose` to be true if `origin == JVMFlagOrigin::ERGONOMIC)`.

Reply:
I've forced `verbose` to be true in another code path. I added the same logic here to be complete.